template-coq/src/ast_denoter.ml

```diff
@@ -201,22 +201,25 @@ struct
       Univ.Level.make (Univ.Level.UGlobal.make dp "" idx)
     | Universes0.Level.Var n -> Univ.Level.var (unquote_int n)
 
-  let unquote_level_expr (trm : Universes0.Level.t * quoted_int) : Univ.Universe.t =
+  let unquote_level_expr (trm : Universes0.Level.t * quoted_int) : Sorts.t =
     let l = unquote_level (fst trm) in
-    let u = Univ.Universe.make l in
+    let u = Sorts.sort_of_univ @@ Univ.Universe.make l in
     let n = unquote_int (snd trm) in
-    if n > 0 && not (Univ.Level.is_prop l) then Univ.Universe.super u
+    if n > 0 && not (Univ.Level.is_prop l) then Sorts.super u
     else u
 
+  let sort_sup s1 s2 =
+    Sorts.sort_of_univ (Univ.Universe.sup (Sorts.univ_of_sort s1) (Sorts.univ_of_sort s1))
```

Comment:
```suggestion
    Sorts.sort_of_univ (Univ.Universe.sup (Sorts.univ_of_sort s1) (Sorts.univ_of_sort s2))
```
if we had this in coq we would only make this typo once

Reply:
Yeah yeah, but this is scaffolding. This function shouldn't exist for sorts in the first place.